aggregator/rollup_broadcaster.go

```diff
@@ -138,10 +221,99 @@ func NewRollupBroadcaster(ctx context.Context, rollupsInfo map[uint32]config.Rol
 		writers = append(writers, writer)
 	}
 
-	return &RollupBroadcaster{
+	broadcaster := &RollupBroadcaster{
 		writers:   writers,
+		logger:    logger,
 		errorChan: make(chan error),
-	}, nil
+	}
+
+	mainnetNextOperatorSetUpdateId, err := avsReader.GetNextOperatorSetUpdateId(ctx)
+	if err != nil {
+		logger.Error("Error fetching operator set update id", "err", err)
+		return nil, err
+	}
+
+	if mainnetNextOperatorSetUpdateId == 0 {
+		go broadcaster.initializeRollupOperatorSetsOnUpdate(ctx, avsReader, avsSubscriber)
+	} else {
+		for _, writer := range broadcaster.writers {
+			go broadcaster.tryInitializeRollupOperatorSet(ctx, writer, mainnetNextOperatorSetUpdateId, avsReader, avsSubscriber)
+		}
+	}
+
+	return broadcaster, nil
+}
+
+func (b *RollupBroadcaster) initializeRollupOperatorSetsOnUpdate(ctx context.Context, avsReader chainio.AvsReaderer, avsSubscriber chainio.AvsSubscriberer) {
+	var operatorSetUpdatedChan chan *opsetupdatereg.ContractSFFLOperatorSetUpdateRegistryOperatorSetUpdatedAtBlock
+
+	avsSubscriber.SubscribeToOperatorSetUpdates(operatorSetUpdatedChan)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case event := <-operatorSetUpdatedChan:
+			b.logger.Info("Received operator set update", "id", event.Id)
+
+			operators, err := avsReader.GetOperatorSetById(ctx, event.Id)
+			if err != nil {
+				b.errorChan <- err
+				continue
+			}
+
+			convertedOperators := make([]registryrollup.RollupOperatorsOperator, len(operators))
+			for i, op := range operators {
+				convertedOperators[i] = registryrollup.RollupOperatorsOperator{
+					Pubkey: registryrollup.BN254G1Point{X: op.Pubkey.X, Y: op.Pubkey.Y},
+					Weight: op.Weight,
+				}
+			}
+
+			for _, writer := range b.writers {
+				go func(writer *RollupWriter) {
+					err := writer.InitializeOperatorSet(ctx, convertedOperators, event.Id)
+					if err != nil {
+						b.errorChan <- err
+					}
+				}(writer)
+			}
+		}
+	}
+}
+
+func (b *RollupBroadcaster) tryInitializeRollupOperatorSet(ctx context.Context, writer *RollupWriter, mainnetNextOperatorSetUpdateId uint64, avsReader chainio.AvsReaderer, avsSubscriber chainio.AvsSubscriberer) {
+	nextOperatorUpdateId, err := writer.sfflRegistryRollup.NextOperatorUpdateId(&bind.CallOpts{})
+	if err != nil {
+		b.logger.Error("Error fetching NextOperatorUpdateId", "err", err)
+		b.errorChan <- err
+		return
+	}
+
+	if nextOperatorUpdateId == 0 {
```

Comment:
Minor, but I would propose to make it even better with early return statement.

```
	if nextOperatorUpdateId != 0 {
             return
	}

       // logic
```

Reply:
Agreed, done in [b06b38d](https://github.com/NethermindEth/near-sffl/pull/68/commits/b06b38dd2c6e3f5336fac1f80a4ea55ebaa7e61d)